apis/standard.oam.dev/v1alpha1/rollout_state.go

```diff
@@ -265,9 +265,12 @@ func (r *RolloutStatus) StateTransition(event RolloutEvent) {
 	}
 	// special handle modified event here
 	if event == RollingModifiedEvent {
+		if r.RollingState == RolloutDeletingState {
+			panic(fmt.Errorf(invalidRollingStateTransition, rollingState, event))
```

Comment:
Does it necessary to panic? Can we set some state that represent this `panic`? My concern is the approllout panic will affect the whole kubevela controller

Reply:
panic makes sense in terms of state transition but it is a problem when all controllers run in the same executable